app/controllers/application_controller.rb

```diff
@@ -10,6 +11,26 @@ class ApplicationController < ActionController::Base
 
   add_flash_types :alert_with_description, :alert_with_items, :confirmation, :tried_to_publish, :tried_to_preview
 
+  #rescue_from Edition::IntegrityAssertionError do |e|
+    #Rails.logger.error(e.message)
+    #redirect_to document_path(params[:document])
+  #end
+
+  rescue_from Assertions::ErrorWithEdition do |e|
+    Rails.logger.error(e.message)
+    redirect_to document_path(e.edition.document)
+  end
```

Comment:
I feel this rescue is too generalised - would there be edition errors where we didn't want to redirect to document_path?